waku/node/peer_manager/peer_manager.nim

```diff
@@ -331,6 +333,11 @@ proc onConnEvent(pm: PeerManager, peerId: PeerID, event: ConnEvent) {.async.} =
     of ConnEventKind.Disconnected:
       discard
 
+proc peerHasProtocol(
+    peerManager: PeerManager, peerId: PeerId, protocol: string
+): bool {.inline, raises: [].} =
+  return peerManager.peerStore[ProtoBook][peerId].contains(protocol)
+
```

Comment:
I would put this in `waku_peer_store.nim` instead.

Or maybe use https://github.com/waku-org/nwaku/blob/0894f0cfea9fac17348607cb50768e17bd788660/waku/node/peer_manager/waku_peer_store.nim#L112